src/Toolkit/Xml.php

```diff
@@ -421,8 +421,7 @@ public static function value($value): ?string
             return $value;
         }
 
-        // TODO: in 3.7.0 use ENT_NOQUOTES | ENT_XML1 instead
-        $encoded = htmlentities($value, ENT_COMPAT);
+        $encoded = htmlentities($value, ENT_QUOTES | ENT_SUBSTITUTE);
```

Comment:
Why the different flags compared to the TODO comment?

Reply:
With `ENT_NOQUOTES | ENT_XML1` unit tests failed. Then I googled again and apparently for `htmlentities`  `ENT_QUOTES | ENT_SUBSTITUTE` is the recommended replacement. So I must've slipped up when writing the comment in the first place.

Reply:
https://php.watch/versions/8.1/html-entity-default-value-changes

Reply:
I traced it back to this discussion: https://github.com/getkirby/kirby/pull/3979#discussion_r763350909

Since the line is used specifically to check if a `CDATA` block is required, we only need to escape characters that are harmful as XML character data outside of CDATA blocks. Quotes are not harmful there (only in attributes), so we don't need to escape them. Whether we use `ENT_SUBSTITUTE` or not doesn't matter as invalid code unit sequences will in any case return a different `$encoded` value. What we do need is `ENT_XML1` as we are encoding specifically for XML output.

So I think our unit tests are wrong and need to be updated. I see two failing ones:

```txt
2) Kirby\Toolkit\XmlTest::testTag
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<name foo="bar"><![CDATA[Süper Önencœded ßtring]]></name>'
+'<name foo="bar">Süper Önencœded ßtring</name>'

kirby/tests/Toolkit/XmlTest.php:187

3) Kirby\Toolkit\XmlTest::testValue with data set #8 ('töst', '<![CDATA[töst]]>')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<![CDATA[töst]]>'
+'töst'

kirby/tests/Toolkit/XmlTest.php:205
```

Both these strings don't actually require `CDATA`. If we change the strings to `String with <not> a tag & some text`, the tests will pass again.

Reply:
I'm rather lost with the `XmlTest` test suite - and what can be changed or not without losing an important test case. So I'd rely there on your assessment.

Reply:
This would be one of the failing tests

```
$tag = Xml::tag('name', 'Süper Önencœded ßtring', ['foo' => 'bar']);
$this->assertSame('<name foo="bar"><![CDATA[Süper Önencœded ßtring]]></name>', $tag);
```

Not sure I completely understand your suggestion how to change it when using `ENT_NOQUOTES | ENT_XML1`.

Reply:
```php
$tag = Xml::tag('name', 'String with <not> a tag & some text', ['foo' => 'bar']);
$this->assertSame('<name foo="bar"><![CDATA[String with <not> a tag & some text]]></name>', $tag);
```

Reply:
And `XmlTest::valueProvider()` would be:

```php
    return [
            [true, 'true'],
            [false, 'false'],
            [1, '1'],
            [null, null],
            ['', null],
            ['<![CDATA[test]]>', '<![CDATA[test]]>'],
            ['<![CDATA[String with <not> a tag & some text]]>', '<![CDATA[String with <not> a tag & some text]]>'],
            ['test', 'test'],
            ['String with <not> a tag & some text', '<![CDATA[String with <not> a tag & some text]]>'],
            ['This is a <![CDATA[test]]> with CDATA', '<![CDATA[This is a <![CDATA[test]]]]><![CDATA[> with CDATA]]>'],
            ['te]]>st', '<![CDATA[te]]]]><![CDATA[>st]]>'],
            ['tö]]>st', '<![CDATA[tö]]]]><![CDATA[>st]]>']
        ];
```

Reply:
Implemented, thanks!